mobile/lib/resources/fetch_data.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+/// Fetches the data from provided source [url]
+/// and bundles it into a form of an iterable
+/// of a provided type [T].
+///
+/// The [fromJson] has to be a function
+/// that converts the provided JSON data to
+/// an object of type [T].
+/// For example, it can be [T]'s fromJson()
+/// constructor.
+///
+/// Throws [TimeoutException] if the fetching
+/// function exceeds given [timeout].
+Future<Iterable<T>> fetchData<T>(
```

Comment:
I'd create get post methods for GET and POST

Reply:
Good idea for the future, I will create the tasks for them